app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -139,6 +139,38 @@ def undeploy_middleware_deployment(ems_ref)
       run_generic_operation(:Undeploy, ems_ref)
     end
 
+    def restart_middleware_server(ems_ref)
+      run_generic_operation(:Shutdown, ems_ref, :restart => true)
+    end
+
+    def shutdown_middleware_server(ems_ref, _params)
+      timeout = 10 # we default to 10s until we get the UI params. params.fetch ':timeout'
+      run_generic_operation(:Shutdown, ems_ref, :restart => false, :timeout => timeout)
+    end
+
+    def suspend_middleware_server(ems_ref, params)
+      timeout = params.fetch ':timeout'
```

Comment:
Should you handle missing params since this is a public method?

Either:
```ruby
timeout = params.fetch(':timeout') || 0
```

Or, 
```ruby
timeout = params.fetch ':timeout'
raise "Missing timeout parameter' unless timeout
```

Something along those lines...